beacon_chain/state/state_transition.py

```diff
@@ -348,3 +347,18 @@ def compute_state_transition(parent_state: Tuple[CrystallizedState, ActiveState]
     )
 
     return crystallized_state, active_state
+
+
+def get_recalculated_states(
```

Comment:
Maybe the name should be more specific as this seems to just handle cycle transitions. Maybe something like `compute_cycle_transitions`?